examples/plot_pp_study.py

```diff
@@ -0,0 +1,362 @@
+
+y_pred = classifier.predict(data_without_crimerate)
+# %%
+#
+# **3. We create bins for LSTAT (colored blue in Figure A above), to understand the models performance on different LSTAT value intervals.**
+#
+#  While we calculate and print Mean, Median, Max, and Min LSTAT
+#  values, the chosen bins are based on the observe Min and Max LSTAT
+#  values.
+#
+lstat = bostonDS.data['LSTAT']
+mean_lstat = mean(lstat)
+median_lstat = median(lstat)
+max_lstat = max(lstat)
+min_lstat = min(lstat)
+
+print('Mean LSTAT ', mean_lstat)
+print('Median LSTAT ', median_lstat)
+print('Max LSTAT ', max_lstat)
+print('Min LSTAT ', min_lstat)
+
+bins = [0, 10, 20, 30, 40]
+print(bins)
+labels = [1, 2, 3, 4]
+
+lstat_binned = pd.cut(lstat, bins, labels=labels)
+# %%
+#
+# **4. We treat lstat_binned (defined in step 3 above) as the sensitive feature on which to assess fairness metrics.**
+#
+accuracy_frame = MetricFrame(accuracy_score, y_true, y_pred, sensitive_features=lstat_binned)
+accuracy_by_group = accuracy_frame.by_group
+
+selection_frame = MetricFrame(selection_rate, y_true, y_pred, sensitive_features=lstat_binned)
+selection_by_group = selection_frame.by_group
+
+print('Selection rate statistics')
+print(selection_frame.overall)
+print(selection_by_group)
+
+print('Accuracy rate statistics')
+print(accuracy_frame.overall)
+print(accuracy_by_group)
+# %%
+#
+FairlearnDashboard(sensitive_features=lstat_binned, sensitive_feature_names=['lstat'], y_true=y_true, y_pred={"initial model": y_pred})
+# %%
+#
+# Fairlearn Results
+# -----------------
+#
+# Upon running Fairlearn with lstat_binned (described in Background) as
+# our sensitive feature, we observe the following disparities in accuracy
+# across bins, and selection rates.
+#
+# .. image:: https://i.imgur.com/x7zJlcU.png
+#    :target: https://i.imgur.com/x7zJlcU.png
+#    :alt:
+#
+# *Figure B: Disparity in accuracy across lstat bins*
+#
+#
+# .. image:: https://i.imgur.com/6Ar1pyS.png
+#    :target: https://i.imgur.com/6Ar1pyS.png
+#    :alt:
+#
+# *Figure C: Disparity in selection rate across lstat bins*
+#
+# Here are some key questions to think about given the results above:
+#
+#
+# * Using mean or median crime rate as the threshold value, the overall
+#   accuracy of the model is 95%. Does that mean the model is good to go?
+#   Can there still be issues despite what seems like a properly working
+#   model? Tradeoffs between accuracy and fairness?
+#
+# * What does it mean that the last bin has the highest selection rate?
+#   Given the sociotechnical context, what would this imply in terms of
+#   the societal impact and outcome?
+#
+# * What would a ‘mitigation’ mean in this scenario? Assuming no
+#   constraints on the mitigations offered in Fairlearn modules, what type
+#   of disparity would we be looking to reduce? (Explored in the section
+#   to follow)
+#
+# Another important question to consider: should the described problem
+# (policing) be framed as an ML task at all?
```

Comment:
This feels like a pretty key question, and something I think we should either raise at the outset of the scenario, or address more prominently here...